scanner/families/malware/yara/yara.go

```diff
@@ -91,7 +94,7 @@ func (s *Scanner) Run(sourceType utils.SourceType, userInput string) error {
 		defer cleanup()
 
 		// Define the yara args to run
-		args := []string{"-C", s.compiledRuleFile, "-r", "-w", "-m", fsPath}
+		args := []string{"-C", s.compiledRuleFile, "-w", "-m", "-a", strconv.Itoa(s.config.PerFileTimeout)}
```

Comment:
```suggestion
		args := []string{"-C", s.compiledRuleFile, "-r", "-w", "-m", "--threads", strconv.Itoa(runtime.NumCPU())}
```
this way, we can remove `MaxGoRoutineNumber` since we have the actual runtime value